src/mtev_rest.c

```diff
@@ -274,7 +274,7 @@ mtev_http_find_matching_route_rule(mtev_http_rest_closure_t *restc)
     int cnt;
     if (mtev_http_is_websocket(restc->http_ctx) == mtev_true) {
       if (strcmp(rule->method, "WS")) continue;
-      if (rule->websocket_protocol == NULL || protocol == NULL || strcmp(rule->websocket_protocol, protocol)) continue;
+      if (rule->websocket_protocol != NULL && protocol != NULL && strcmp(rule->websocket_protocol, protocol)) continue;
```

Comment:
It looks like you were applying De Morgan's here but you did not negate the `strcmp` at the end.  Should there be a `!` in front of that, or was that in purpose?

Reply:
it was in purpose.  This was a correction, so the new statement is different than the old.